clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -404,6 +411,12 @@ private void maybeTriggerWakeup() {
             throw new WakeupException();
         }
     }
+    
+    private void maybeThrowInterruptException() {
```

Comment:
On your question about why we catch `WakeupException` in those snippets, it's because of the rebalance callback. Users will often invoke `commitSync()` from with the revocation handler, for example, or they might call one of the `seek` APIs in the assignment handler.